core/divefilter.cpp

```diff
@@ -14,24 +14,26 @@
 #endif
 
 // Set filter status of dive and return whether it has been changed
-bool DiveFilter::setFilterStatus(struct dive *d, bool shown) const
+bool DiveFilter::setFilterStatus(struct dive *d, bool shown,
+				 const std::vector<dive *> selection, std::vector<dive *> &removeFromSelection) const
 {
 	bool old_shown, changed;
 	if (!d)
 		return false;
 	old_shown = !d->hidden_by_filter;
 	d->hidden_by_filter = !shown;
 	if (!shown && d->selected)
-		deselect_dive(d);
+		removeFromSelection.push_back(d);
 	changed = old_shown != shown;
 	if (changed)
 		shown_dives += shown - old_shown;
 	return changed;
 }
 
-void DiveFilter::updateDiveStatus(dive *d, bool newStatus, ShownChange &change) const
+void DiveFilter::updateDiveStatus(dive *d, bool newStatus, ShownChange &change,
+				  const std::vector<dive *> selection, std::vector<dive *> &removeFromSelection) const
```

Comment:
again, both vectors as argument, and `selection` is only passed to `setFilterStatus()` which doesn't need it.

At this point I'm guessing that this will make sense a few more commits down the road, maybe?

Reply:
fixed.